### PR TITLE
(feat) Update location picker design

### PR DESCRIPTION
**Pull request by denniskigen**
## Requirements
- [x] This PR has a title that briefly describes the work done, including the ticket number if there is a ticket.

#### For changes to apps
- [x] My work conforms to the [**OpenMRS 3.0 Styleguide**](https://om.rs/styleguide).

#### If applicable
- [ ] My work includes tests or is validated by existing tests.
- [ ] I have updated the [esm-framework mock](https://github.com/openmrs/openmrs-esm-core/blob/main/packages/framework/esm-framework/mock.tsx) to reflect any API changes I have made.

## Summary

Updates the location picker UI to match the [available designs](https://app.zeplin.io/project/60d59321e8100b0324762e05/screen/6220c7c14619b2570b4b23d3/).

## Screenshots

> Before
Screenshot 2022-09-13 at 22 58 51

> After
Screenshot 2022-09-13 at 22 57 50

> Empty state 
Screenshot 2022-09-14 at 00 19 09

## Related Issue

https://issues.openmrs.org/browse/O3-1295

### CODE REVIEW

File: packages/apps/esm-login-app/src/location-picker/location-picker.component.tsx [hunk 1]
```diff
@@ -149,80 +149,68 @@ const LocationPicker: React.FC<LocationPickerProps> = ({
...
-                    </RadioButtonGroup>
-                  )}
-                  {locationData?.length === 0 && (
-                    <p className={styles.locationNotFound}>
-                      {t(
-                        "locationNotFound",
-                        "Sorry, no matching location was found"
-                      )}
-                    </p>
+          <>
```
Comment (ZacButko) [inline packages/apps/esm-login-app/src/location-picker/location-picker.component.tsx:152]: I don't see why this needs to be in a fragment?

File: packages/apps/esm-login-app/src/location-picker/location-picker.scss [hunk 2]
```diff
@@ -11,13 +11,16 @@
   width: 100vw;
 }
 
+.paddedContainer {
+  padding: 1.5rem;
```
Comment (ZacButko) [inline packages/apps/esm-login-app/src/location-picker/location-picker.scss:15]: spacing-06

File: packages/apps/esm-login-app/src/location-picker/location-picker.scss [hunk 3]
```diff
@@ -38,9 +37,10 @@
 
 .searchResults {
   @extend .bodyShort02;
-  margin: 1rem 0rem;
   max-height: 100%;
   overflow-y: auto;
+  padding: 0rem 1.5rem;
```
Comment (ZacButko) [inline packages/apps/esm-login-app/src/location-picker/location-picker.scss:42]: spacing-06

File: packages/apps/esm-login-app/src/location-picker/location-picker.scss [hunk 4]
```diff
@@ -52,6 +52,7 @@
   display: flex;
   overflow-y: auto;
   margin-top: 0.5rem;
+  padding: 0rem 0.125rem 0rem;
```
Comment (ZacButko) [inline packages/apps/esm-login-app/src/location-picker/location-picker.scss:55]: spacing-01

File: packages/apps/esm-login-app/src/location-picker/location-picker.scss [hunk 5]
```diff
@@ -65,12 +66,13 @@
 }
 
 .locationNotFound {
-  @extend .bodyShort01;
-  padding: 1rem 0rem;
-  color: $color-gray-70;
+  @extend .productiveHeading01;
+  color: $text-02;
+  margin-bottom: 1rem;
```
Comment (ZacButko) [inline packages/apps/esm-login-app/src/location-picker/location-picker.scss:71]: spacing-05

File: packages/apps/esm-login-app/src/location-picker/location-picker.scss [hunk 6]
```diff
@@ -65,12 +66,13 @@
...
 }
 
 .confirmButton button {
+  margin: 1.5rem;
```
Comment (ZacButko) [inline packages/apps/esm-login-app/src/location-picker/location-picker.scss:75]: spacing-06

File: packages/apps/esm-login-app/src/location-picker/location-picker.scss [hunk 7]
```diff
@@ -108,3 +110,14 @@ pagination > button:first-child:hover {
   align-items: center;
   margin-top: spacing.$spacing-05;
 }
+
+.loadingContainer {
+  margin-top: 1.5rem;
```
Comment (ZacButko) [inline packages/apps/esm-login-app/src/location-picker/location-picker.scss:115]: spacing-06

File: packages/apps/esm-login-app/src/location-picker/location-picker.scss [hunk 8]
```diff
@@ -108,3 +110,14 @@ pagination > button:first-child:hover {
...
+}
+
+.emptyState {
+  display: flex;
+  flex-direction: column;
+  justify-content: center;
+  margin: 1.5rem auto 0rem;
```
Comment (ZacButko) [inline packages/apps/esm-login-app/src/location-picker/location-picker.scss:122]: spacing-06

File: packages/apps/esm-login-app/translations/en.json [hunk 9]
```diff
@@ -5,22 +5,16 @@
   "contactAdmin": "Contact the site administrator",
   "continue": "Continue",
   "error": "Error",
-  "found": "found",
   "invalidCredentials": "Invalid username or password",
   "loading": "Loading",
-  "locationNotFound": "Sorry, no matching location was found.",
-  "locations": "locations",
```
Comment (ZacButko) [inline packages/apps/esm-login-app/translations/en.json:12]: I can't view the design doc you linked to, but a little sad to see the "viewing {x} of {total} locations" feature go away

File: packages/apps/esm-login-app/src/location-picker/location-picker.component.tsx [hunk 10]
```diff
@@ -149,80 +149,68 @@ const LocationPicker: React.FC<LocationPickerProps> = ({
...
+              labelText={t("searchForLocation", "Search for a location")}
+              id="search-1"
+              placeholder={t("searchForLocation", "Search for a location")}
+              onChange={(event) => search(event.target.value)}
+              name="searchForLocation"
+              size="lg"
+            />
+            <div
+              className={styles.searchResults}
+              style={{ height: `${chooseLocation.numberToShow * 2.875}rem` }}
```
Comment (ZacButko) [inline packages/apps/esm-login-app/src/location-picker/location-picker.component.tsx:164]: I know this wasn't you, but it's a little weird

**Review by ZacButko — APPROVED** [review]
Some css could better conform to carbon's predefined spacing options, and I'm not able to see the link to original designs to confirm, but happy for style cleanup.

File: packages/apps/esm-login-app/src/location-picker/location-picker.component.tsx (hunk 10 above)
Comment (denniskigen) [reply packages/apps/esm-login-app/src/location-picker/location-picker.component.tsx:164]: Per the [design](https://app.zeplin.io/project/60d59321e8100b0324762e05/screen/6220c7c2c7f55a5949a714e6), looks like the container should have a height of `408px`.

File: packages/apps/esm-login-app/translations/en.json (hunk 9 above)
Comment (denniskigen) [reply packages/apps/esm-login-app/translations/en.json:12]: Something I'd be interested to hear Ciaran's thoughts about.

**Comment by denniskigen** [comment]
Is it good to go, @ZacButko?

**Review by ZacButko — APPROVED** [review]

**Pull request merged**